ludwig/schema/features/augmentation/utils.py

```diff
@@ -0,0 +1,157 @@
+from dataclasses import field
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import TYPE
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.features.augmentation.base import BaseAugmentationContainerConfig
+from ludwig.utils.registry import Registry
+
+
+@DeveloperAPI
+@dataclass(repr=False)
+class ImageAugmentationContainerConfig(BaseAugmentationContainerConfig):
+    """Augmentation container for image features."""
+
+    pass
+
+
+# TODO: Is all of this needed?
+# augmentation_registry = Registry()
+#
+#
+# @DeveloperAPI
+# def register_augmentation(name: str):
+#     def wrap(augmentation_config: BaseAugmentationConfig):
+#         augmentation_registry[name] = augmentation_config
+#         return augmentation_config
+#
+#     return wrap
+
+_augmentation_config_registry = Registry()
+
+
+@DeveloperAPI
+def get_augmentation_config_registry() -> Registry:
+    return _augmentation_config_registry
+
+
+def register_augmentation_config(name: str):
+    def wrap(cls):
+        get_augmentation_config_registry()[name] = cls
+        return cls
+
+    return wrap
+
+
+def get_augmentation_cls(name: str):
+    return get_augmentation_config_registry()[name]
+
+
+@DeveloperAPI
+def AugmentationContainerDataclassField(default=None, description=""):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify an augmentation
+    config.
+
+    Args:
+        default: The default augmentation config to use.
+        description: The description of the augmentation config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to an augmentation config.
+    """
+
+    class AugmentationContainerMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a list for a valid augmentation config from the
+        augmentation_registry and creates a corresponding JSON schema for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            if value is None:
+                return None
+            augmentation_list = []
+            for augmentation in value:
+                augmentation_op = augmentation[TYPE]
+                augmentation_cls = get_augmentation_cls(augmentation_op)
+                pre = augmentation_cls()
+                try:
+                    augmentation_list.append(pre.Schema().load(augmentation))
+                except (TypeError, ValidationError) as error:
+                    raise ValidationError(
+                        f"Invalid augmentation params: {value}, see `{pre}` definition. Error: {error}"
+                    )
+            return augmentation_list
+
+        @staticmethod
+        def _jsonschema_type_mapping():
+            return get_augmentation_jsonschema()
+
+    try:
+        augmentation_container = ImageAugmentationContainerConfig
```

Comment:
These three lines should copy what you are doing above in `_deserialize` except you do everything to `default` instead of `value`. 

Some context on this pattern:
The point of defining `dump_default` and `load_default` using a "`Schema.load`" is so that the `default` argument (which is defined at schema declaration) _is also validated_. Hypothetically, you could delete these lines entirely and set `dump_default=default` directly and `marshmallow` would accept that - we _don't_ want that to happen (marshmallow only allows it because it is designed to be as flexible as possible), so we duplicate the validation logic we use for user-specified values.

I bet this might be causing some of your test weirdness. Because what is currently happening is you use `ImageAugmentationContainerConfig` to `load` the `default` argument provided to this field - and `ImageAugmentationContainerConfig`'s schema is empty so by using `Schema().load` on empty dictionaries `load_default` gets set to `ImageAugmentationConfig()` (an instance of that empty class instead of what you want, which is a list of augmentation instances) and `dump_default` gets set to `{}`.

Reply:
@ksbrar   I'd appreciate if you take a look at how I interepreted your comments, especially how I assigned values to `load_default` and `dump_default`and set `default_factory` parameter in the `field()` call.